src/inmanta/agent/executor.py

```diff
@@ -75,7 +75,7 @@ class DryrunResult:
     messages: list[LogLine]
 
 
-class ResourceDetails:
+class ResourceSpec:
```

Comment:
rename 1.

motivation: conflict with `deploy.state.ResourceDetails`.

notes:
- feels somewhat clunky. Then again, so does `ResourceDetails`.
- I also renamed `deploy.state.ResourceDetails`, so this rename here is not strictly necessary. Let's only keep it if we consider it an improvement even apart from the name conflict.

Reply:
It is the same as the ResourceIntent, but in a different domain. 

We don't really have a convention for it, but it is not weird for the same concept to have the same name.

Reply:
I think we have to add a prefix or suffix to indicate to indicate the domain: SchedulerResourceIntent and ExecutorResourceIntent. Then it's clear they represent something similar and at the same time it's easy to keep them apart.

Reply:
> It is the same as the ResourceIntent, but in a different domain.

I partially agree, but on the other hand it also contains a bunch of metadata that could be argued to be outside of the resource's own intent, like the model version and requires. That's why I don't find intent very appropriate here, while I find it perfectly appropriate for the other one.

> I think we have to add a prefix or suffix to indicate the domain: SchedulerResourceIntent and ExecutorResourceIntent. Then it's clear they represent something similar and at the same time it's easy to keep them apart.

I personally don't like this. I agree that context is occasionally important. But when it is, I'm strongly of the opinion that it should be provided by qualified imports, not name prefixing.

Reply:
> I'm strongly of the opinion that it should be provided by qualified imports, not name prefixing.

The problem with using qualified imports is that it doesn't make you actively aware about the fact that there is another class with exactly the same name somewhere in the code base. So using qualified imports to make the distinction doesn't make it easier for someone to enter the code base.

Reply:
I'm still inclined to leave this one at `ResourceDetails`. It's not perfect, but it does describe what it is, i.e. all details required for a resource action on a given resource.

The scheduler on the other hand deals a lot more directly with intent, irrespective of model version or requires-provides. The name change that we agreed in in that domain should address Arnaud's naming concerns if we leave this one as is.

@wouterdb how strongly do you feel we should rename this one?

Reply:
The problem with the qualified imports approach is that I bet that someone in the team will introduce a PR in the coming 6 months (if it even takes that long) that is not using the qualified imports for this case. It's an approach that works if you work alone on a project but it doesn't work when there is a team of developers involved.

Reply:
I agree there's a risk to that, but I also think we should simply pay more attention to imports, and we have been growing better in that regard over the past year or so. I'd love to continue this discussion, in part *because* it's something I've been thinking about a lot recently (and discussing with various people at the office), but perhaps this is not the time or place, considering that we have a proposal that gets rid of the ambiguity here? Unless you don't like that proposal?

Reply:
I don't feel very strong about this at all.

Reply:
It's not perfect. But it's better than two classes with the same name. So it's fine by me.